### DISCUSSION TITLE
Guake 3.8

### DISCUSSION THREAD

**Discussion by Davidy22** [Guake/guake#1904] [Ideas]
It's been a long while since last release and we've gotten some hefty features/fixes in. The older pull requests are mostly cleaned out now, just some that need work are left. I do particularly want to get #1897 in to finish dealing with something that looks like it's been a heavy blocker for package maintainers, wouldn't mind getting some of my other pull requests in as well. Would like to have all the things we're going to have in the next version merged in for a little while and frozen so we can use it day-to-day to make sure 3.8's clean, a release after this amount of time might get attention. Don't have access to any keys and I probably don't know all the channels guake has to get a new package out, but I do want to get the [currently unofficial ppa](https://launchpad.net/~linuxuprising/+archive/ubuntu/guake) on board with us so we can give people who want ppa a link ourselves, and polish off the highest voted request on feathub.

**Comment by gsemet** [comment]
Usually, packager acts on their own will, so when I release on pypi the package are updated a few weeks later. some distrib like debian has weird policy (they never update a package except patch fix, so guake 3.8 won't be available in already released versions).

I would like the following actions to be made before release, tell me if you can do it:
- update all dependencies (`make update`), and also update pipenv pip etc in the build environment. I do not have a linux env so i cannot do it easily
- some tests of course

on it is ok, I can do the release and push to pypi. (should be automatic through the pipeline, I will check it).

also, I would like to recommend installing guake through pipx, I hope it works, but I cannot confirm. Can you test if old version of guake works with pipx?

**Reply by Davidy22** [reply]
Yeah I figure debian's going to trickle it out, sooner we get started sooner we'll eventually get our new stuff propagated to users. make update stops before running the second line, not sure what's going on with that but I'm running the second half of it now. I can freeze dev work and just use what's in git master day-to-day, don't have a ton of other people to do the same but maybe after we release we might get some attention and volunteers for it for future. So far, I've noticed a graphical issue #1894 although it's probably not blocking.

I did remove a pypi related thing a bit ago which might mean that it's not automatic in the pipeline anymore, also the key needs to be regenerated for safety, because it was just sitting there in plaintext. Poetry does have a nice easy deploy option that also does it in a way that doesn't commit any keys into the repo, so that's a thing I could do while I'm soft-banning myself from changing code.

**Reply by Davidy22** [reply]
Update on pipx and make update, during the make update I replaced dephell with flakehell because dephell discontinued some time ago and had pip pinned to an ancient version, flakehell's got some things to say about the code that I'll get to mopping up. Guake does not work when installed through pipx, because of the gi dependency that needs to be installed through not-pip. Can look further for a pure pip substitute, hope is slim though.

**Reply by Davidy22** [reply]
Alright, the dependency update's been done, now just dogfooding. pipx still not recommendable because of the dependency outside of the python ecosystem.

**Reply by mlouielu** [reply]
How about `poetry`?

They also provide `poetry publish` and `poetry publish -r testpypi` for publising.

**Reply by gsemet** [reply]
Thanks a lot for the update. For the poetry migration, this is not too difficult but we can do it after the next release.

**Reply by mlouielu** [reply]
Sure, no rush on this.

**Reply by mlouielu** [reply]
I've fix #1894 by #1907, if merged then +1 for 3.8 release.
